### PR TITLE
[SPARK-15087][CORE][SQL] Remove AccumulatorV2.localValue and keep only value

### PR DESCRIPTION
**Pull request by techaddict**
## What changes were proposed in this pull request?

Remove AccumulatorV2.localValue and keep only value
## How was this patch tested?

existing tests

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/AccumulatorV2.scala [hunk 1]
```diff
@@ -128,21 +128,7 @@ abstract class AccumulatorV2[IN, OUT] extends Serializable {
   /**
    * Access this accumulator's current value; only allowed on driver.
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/AccumulatorV2.scala]: need to update this doc too

**Comment by techaddict** [comment]
@cloud-fan fixed 👍

File: core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala [hunk 2]
```diff
@@ -126,23 +126,9 @@ abstract class AccumulatorV2[IN, OUT] extends Serializable {
   def merge(other: AccumulatorV2[IN, OUT]): Unit
 
   /**
-   * Access this accumulator's current value; only allowed on driver.
+   * Defines the current value of this accumulator
    */
-  final def value: OUT = {
-    if (atDriverSide) {
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala]: We should put this check in the old accumulator, to avoid breaking its behaviour.

Comment (techaddict) [inline core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala]: done 👍

**Comment by HyukjinKwon** [comment]
(This is a pretty minor but I think `cc @rxin` can be removed but in the comments because the PR description explains the PR itself and the names of reviewers might not be related with the PR itself. `@` will be removed by Python merge script when merging but `cc` and `mengxr` will still remain, eg. https://github.com/apache/spark/commit/4514aebd1e807a665c270bfdc3f1127b3a1da898. Also I remember I was told so by one of committers)

**Comment by cloud-fan** [comment]
LGTM, cc @rxin for another look

**Comment by rxin** [comment]
Thanks - merging in master.

File: core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala [hunk 3]
```diff
@@ -394,7 +394,7 @@ private[spark] class TaskSchedulerImpl(
         // deserialized.  This brings trouble to the accumulator framework, which depends on
         // serialization to set the `atDriverSide` flag.  Here we call `acc.localValue` instead to
```
Comment (tedyu) [inline core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala]: This no longer applies

**Pull request closed without merging**